api/tests/src/auth/test_api_gateway_key_auth.py

```diff
@@ -0,0 +1,135 @@
+    """Test API key validation when key is inactive"""
+    user = UserFactory.create()
+    UserApiKeyFactory.create(user=user, key_id="inactive-key", is_active=False)
+    db_session.commit()
+
+    with pytest.raises(ApiKeyValidationError, match="API key is inactive"):
+        validate_api_key_in_db("inactive-key", db_session)
+
+
+@freeze_time("2024-11-14 12:00:00", tz_offset=0)
+def test_api_gateway_key_auth_happy_path(mini_app, enable_factory_create, db_session):
+    """Test successful API Gateway key authentication"""
+    user = UserFactory.create()
+    UserApiKeyFactory.create(user=user, key_id="valid-gateway-key", is_active=True, last_used=None)
+    db_session.commit()
+
+    resp = mini_app.test_client().get(
+        "/dummy_auth_endpoint", headers={"X-API-Key": "valid-gateway-key"}
+    )
+
+    assert resp.status_code == 200
+    assert resp.get_json()["message"] == "ok"
+
+    # Note: last_used timestamp is updated during authentication but not committed
+    # in the test environment due to transaction management
+
+
+def test_api_gateway_key_auth_invalid_key(mini_app, enable_factory_create, db_session):
+    """Test API Gateway key authentication with invalid key"""
+    resp = mini_app.test_client().get("/dummy_auth_endpoint", headers={"X-API-Key": "invalid-key"})
+
+    assert resp.status_code == 401
+    assert resp.get_json()["message"] == "Invalid API key"
+
+
+def test_api_gateway_key_auth_inactive_key(mini_app, enable_factory_create, db_session):
+    """Test API Gateway key authentication with inactive key"""
+    user = UserFactory.create()
+    UserApiKeyFactory.create(user=user, key_id="inactive-gateway-key", is_active=False)
+    db_session.commit()
+
+    resp = mini_app.test_client().get(
+        "/dummy_auth_endpoint", headers={"X-API-Key": "inactive-gateway-key"}
+    )
+
+    assert resp.status_code == 401
+    assert resp.get_json()["message"] == "API key is inactive"
+
+
+def test_api_gateway_key_auth_no_key_header(mini_app, enable_factory_create, db_session):
+    """Test API Gateway key authentication with missing header"""
+    resp = mini_app.test_client().get("/dummy_auth_endpoint", headers={})
+
+    assert resp.status_code == 401
+
+
+def test_api_gateway_key_auth_empty_key_header(mini_app, enable_factory_create, db_session):
+    """Test API Gateway key authentication with empty header"""
+    resp = mini_app.test_client().get("/dummy_auth_endpoint", headers={"X-API-Key": ""})
+
+    assert resp.status_code == 401
+    assert resp.get_json()["message"] == "Invalid API key"
+
+
+def test_api_gateway_key_auth_multiple_active_keys(mini_app, enable_factory_create, db_session):
+    """Test that different API keys for the same user work independently"""
+    user = UserFactory.create()
+    UserApiKeyFactory.create(user=user, key_id="user-key-1", is_active=True)
+    UserApiKeyFactory.create(user=user, key_id="user-key-2", is_active=True)
+    db_session.commit()
```

Comment:
The factory create method calls commit every time it runs - no need to commit.

```suggestion
    user = UserFactory.create()
    UserApiKeyFactory.create(user=user, key_id="user-key-1", is_active=True)
    UserApiKeyFactory.create(user=user, key_id="user-key-2", is_active=True)
```